ooniprobe/Utility/UIDevice-DeviceInfo.m

```diff
@@ -0,0 +1,29 @@
+#include <sys/sysctl.h>
+#import "UIDevice-DeviceInfo.h"
+
+@implementation UIDevice (DeviceInfo)
+
+- (NSString *)getSysInfoByName:(const char *)typeSpecifier
+{
+    size_t size = 0;
+    int ret = sysctlbyname(typeSpecifier, NULL, &size, NULL, 0);
+    if (ret == 0 || size == 0) abort();
```

Comment:
Did you test this code?

Reply:
Just tested on the simulator. ret is 0 and crashes the code.

Reply:
maybe we should check the ret value ?

Reply:
removing both if (ret ==0) works

Reply:
`sysctlbyname` returns `0` on success so you should write `ret != 0`

Reply:
oh I see. tested and working